### ISSUE TITLE
Crystal Defenders: Vanguard Storm crash on Android

### ISSUE THREAD

**Issue by thundergun1** [hikari-no-yume/touchHLE#321]
Just tested out Crystal Chronicles Vanguard on the newest nightly build, and it crashes on the first level. When you start the 3rd wave, an NPC talks to you, but the textbox is buggy and doesn't display right. (This could be another adreno issue)

![Screenshot_20240620_225058_touchHLE](https://github.com/touchHLE/touchHLE/assets/133943803/b4f309cb-8c12-4be9-84a9-f0c8a78a9f22)

You can still click the screen which gives you the regular touch prompt sound, though nothing changes until the game crashes, I've attached the log below:

[log.txt](https://github.com/user-attachments/files/15920492/log.txt)

As usual i'm on Android 12 with Adreno GPU

**Comment by omagadsamsa** [comment]
I find fix https://youtu.be/vH4xiK6pLhc?si=6pl9Hg8viqm_KEZM
[com.square-enix.VanguardSM.zip](https://github.com/user-attachments/files/15920551/com.square-enix.VanguardSM.zip)
in touchhle_sandbox

**Comment by thundergun1** [comment]
> I find fix https://youtu.be/vH4xiK6pLhc?si=6pl9Hg8viqm_KEZM [com.square-enix.VanguardSM.zip](https://github.com/user-attachments/files/15920551/com.square-enix.VanguardSM.zip) in touchhle_sandbox

Just tried it, doesn't work

**Comment by thundergun1** [comment]
On another note, the game also crashes when going into the Options menu

**Comment by omagadsamsa** [comment]
> On another note, the game also crashes when going into the Options menu

no no no tap button continue

**Comment by ciciplusplus** [comment]
@thundergun1 thanks for reporting the crashes.
The one about the "Options"  is already known, you can see that in that appdb.
About other one, I need to reproduce on my side. Was it on Android?

@omagadsamsa please stop spamming here, it's not helpful

**Comment by thundergun1** [comment]
> @thundergun1 thanks for reporting the crashes. The one about the "Options" is already known, you can see that in that appdb. About other one, I need to reproduce on my side. Was it on Android?
> 
> @omagadsamsa please stop spamming here, it's not helpful

Yeah, I'm on Android 12 if that makes much difference. And apologies i hadn't checked the appdb, I got the news of the update from your twitter

**Comment by ciciplusplus** [comment]
> > @thundergun1 thanks for reporting the crashes. The one about the "Options" is already known, you can see that in that appdb. About other one, I need to reproduce on my side. Was it on Android?
> > @omagadsamsa please stop spamming here, it's not helpful
> 
> Yeah, I'm on Android 12 if that makes much difference. And apologies i hadn't checked the appdb, I got the news of the update from your twitter

Thanks, not an issue. If you have exact steps to reproduce, it could be helpful.

**Comment by thundergun1** [comment]
> > > @thundergun1 thanks for reporting the crashes. The one about the "Options" is already known, you can see that in that appdb. About other one, I need to reproduce on my side. Was it on Android?
> > > @omagadsamsa please stop spamming here, it's not helpful
> > 
> > 
> > Yeah, I'm on Android 12 if that makes much difference. And apologies i hadn't checked the appdb, I got the news of the update from your twitter
> 
> Thanks, not an issue. If you have exact steps to reproduce, it could be helpful.

As far as I could tell I play the game as normal, but once you get to wave 3 of the first level, an NPC speaks to you. The textbox gets bugged out as seen in the screenshot of the OP, and I tap the screen to proceed through the text, sometimes I hear the prompt that I've clicked and sometimes I don't, and eventually the game just crashes without me getting passed that text box. 

Sorry I can't be more specific, but I did tend to tap the right side of the screen

**Comment by ciciplusplus** [comment]
Cannot reproduce on my side, it's working as expected on my device
![Screenshot_20240621-182701](https://github.com/touchHLE/touchHLE/assets/91161081/ed3df2d1-6608-4446-93e8-3b1ae0ce0b74)
![Screenshot_20240621-182648](https://github.com/touchHLE/touchHLE/assets/91161081/c385cbb8-d298-4abc-9089-9f612dabd239)

**Comment by thundergun1** [comment]
That's odd. Are there multiple versions of the game? Mine is 1.0.0. The only other thing I can think of is the Adreno GPU, since that is probably the cause of the initial bug with the text box

**Comment by omagadsamsa** [comment]
> That's odd. Are there multiple versions of the game? Mine is 1.0.0. The only other thing I can think of is the Adreno GPU, since that is probably the cause of the initial bug with the text box

1.0.1 problem bug

lost media Lite version

**Comment by thundergun1** [comment]
https://github.com/touchHLE/touchHLE/assets/133943803/b256aaa4-5eb0-49f9-8dd1-ef2bde623d3c

So odd update. I went to record the crash but instead somehow it didn't happen this time. Instead the game played but as you can see from the video the bugged textbox remained so it was a pain to play, I did manage to beat it though and am now onto the second level. I wonder if in my blind tapping I somehow clicked options and that's what's been causing the crash all along?

**Comment by omagadsamsa** [comment]
> Recorder_21062024_174752_compressed.mp4 
> So odd update. I went to record the crash but instead somehow it didn't happen this time. Instead the game played but as you can see from the video the bugged textbox remained so it was a pain to play, I did manage to beat it though and am now onto the second level. I wonder if in my blind tapping I somehow clicked options and that's what's been causing the crash all along?

its problem and in New level I checked

**Comment by ciciplusplus** [comment]
@omagadsamsa please stop making low-effort incomprehensible comments here.
If you don't stop, I'll ban you from the repository, sorry.

**Comment by ciciplusplus** [comment]
I can reproduce the same issue on Nexus 5X, Android 8 with `Driver info: OpenGL ES-CM 1.1 / Qualcomm / Adreno (TM) 418`
But not on Pixel 6a with `Mali-G78 MP20` or Xiaomi Redmi 9C with `PowerVR GE8320`

**Comment by thundergun1** [comment]
> I can reproduce the same issue on Nexus 5X, Android 8 with `Driver info: OpenGL ES-CM 1.1 / Qualcomm / Adreno (TM) 418` But not on Pixel 6a with `Mali-G78 MP20` or Xiaomi Redmi 9C with `PowerVR GE8320`

So it's definitely another Adreno issue then?

**Comment by ciciplusplus** [comment]
> So it's definitely another Adreno issue then?

Very probable

I have another report of the same issue from an user: https://twitter.com/yducfhbdgn/status/1804695501712875710
It's on Samsung Galaxy s9, probably Adreno 630 as well

**Comment by thundergun1** [comment]
Adreno issues do seem to be getting more common as more games come out, though Zombie Infection and Dungeon Hunter are safe from what I've played. So far this game, Tony Hawk, RE4, RE Degeneration, MLB World Series and Driver all seem to have issues, though most of them have been minor thankfully.

Mine is a Samsung Galaxy A42
